_api_app/app/Sites/Sections/Entries/Galleries/GalleryLinkRenderService.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Sites\Sections\Entries\Galleries;
+
+use App\Shared\Storage;
+
+class GalleryLinkRenderService extends EntryGalleryRenderService
+{
+    public $entry;
+    public $siteSettings;
+    public $siteTemplateSettings;
+    public $storageService;
+    public $isEditMode;
+
+    public $galleryItemsData;
+    public $galleryItems;
+
+    public function __construct(
+        array $entry,
+        array $siteSettings,
+        array $siteTemplateSettings,
+        Storage $storageService,
+        $isEditMode
+    ) {
+        $this->entry = $entry;
+        $this->siteSettings = $siteSettings;
+        $this->siteTemplateSettings = $siteTemplateSettings;
+        $this->storageService = $storageService;
+        $this->isEditMode = $isEditMode;
+
+        parent::__construct();
+
+        $this->galleryItemsData = $this->getGalleryItemsData();
```

Comment:
WHy not call this method at Render time?

Reply:
You mean to put it in the `render()` method? My initial idea was to put it inside constructor so that it's available for other methods as well (maybe in the future).